dietpi/dietpi-software

```diff
@@ -12097,6 +12104,73 @@ If no WireGuard (auto)start is included, but you require it, please do the follo
 
 			unset -v choice_required domain port invalid_text server_ip
 		fi
+
+		if To_Install 213 # soju
+		then
+			# Dependencies
+			aDEPS=('gcc' 'make' 'pkg-config' 'scdoc' 'libpam-dev' 'libsqlite3-dev')
+
+			# Download
+			local fallback_url='https://github.com/emersion/soju/releases/download/v0.8.0/soju-0.8.0.tar.gz'
+			Download_Install "$(curl -sSfL 'https://api.github.com/repos/emersion/soju/releases/latest' | mawk -F\" '/^ *"browser_download_url": ".*\/soju-[^"\/]*\.tar\.gz"$/{print $4}')"
+
+			# Build
+			G_EXEC cd soju-*
+			G_EXEC_OUTPUT=1 G_EXEC make all "-j$(nproc)" config_path='/mnt/dietpi_userdata/soju/config' GOFLAGS='-tags=pam,libsqlite3'
+
+			# Install
+			local commands=('soju' 'sojuctl' 'sojudb')
+			local manpages=('doc/soju.1' 'doc/sojuctl.1')
+			G_EXEC cp "${commands[@]}" /usr/local/bin/
+			G_EXEC mkdir -p /usr/local/share/man/man1
+			G_EXEC cp "${manpages[@]}" /usr/local/share/man/man1/
+
+			# Config
+			if [[ ! -f '/mnt/dietpi_userdata/soju/config' ]]
+			then
+				# We shall default to on-disk chat logs
+				# It's a lossy format, as stated under `message-store` subsection in https://soju.im/doc/soju.1.html#CONFIG_FILE
+				# But it's the recommended setting as per https://git.sr.ht/~emersion/soju/tree/master/doc/getting-started.md
+				G_EXEC mkdir -p /mnt/dietpi_userdata/soju
+				cat <<- '_EOF_' > /mnt/dietpi_userdata/soju/config
+				db sqlite3 /mnt/dietpi_userdata/soju/data.db
+				message-store fs /mnt/dietpi_userdata/soju/logs
+				listen irc+insecure://
+				listen unix+admin://
+				_EOF_
+			fi
+
+			# User
+			Create_User -d /mnt/dietpi_userdata/soju soju
+			G_EXEC chown 'soju:soju' -R /mnt/dietpi_userdata/soju
+			G_EXEC chmod 0660 -R /mnt/dietpi_userdata/soju
+			G_EXEC chmod 0770 /mnt/dietpi_userdata/soju
+
+			# Service: https://github.com/emersion/soju/blob/master/contrib/soju.service
+			# - CAP_NET_BIND_SERVICE needed for builtin identd server: https://github.com/emersion/soju/blob/master/doc/packaging.md#binding-to-privileged-ports
+			cat << '_EOF_' > /etc/systemd/system/soju.service
+[Unit]
+Description=soju IRC bouncer service
+Documentation=https://soju.im/
+Documentation=man:soju(1) man:sojuctl(1)
+Wants=network-online.target
+After=network-online.target
+
+[Service]
+User=soju
+RuntimeDirectory=soju
+RuntimeDirectoryMode=0770
+RuntimeDirectoryPreserve=restart
+WorkingDirectory=/mnt/dietpi_userdata/soju
+AmbientCapabilities=CAP_NET_BIND_SERVICE
+ExecStart=/usr/local/bin/soju
+ExecReload=/bin/kill -HUP $MAINPID
+Restart=on-failure
```

Comment:
Ah, in infinite restart loop. This needs to be removed, as it is usually just harmful, spams the logs and makes it difficult to find the actual error buried after pages of restart logs.